### PR TITLE
obs-ffmpeg: Enable demuxer options for sources

### PR DESCRIPTION
**Pull request by pkviet**
(Also modifies deps/media-playback.)
This commit enables passing demuxer options to input sources (whether
local files or other inputs, such as streams or devices).
Syntax is the same as for passing  ffmpeg codec or muxer options in obs:
option1=... option2=...
Ex: for a local file (e.g. gif), input option: ignore_loop=1 .
Ex 2: for a device like decklink, input format: decklink ; input option: 
format_cc=Hp30 .
For all relevant options, refer to ffmpeg documentation (sections on
 demuxers & input devices).

### CODE REVIEW

File: deps/media-playback/media-playback/media.c [hunk 1]
```diff
@@ -541,15 +543,21 @@ static bool init_avformat(mp_media_t *m)
...
 	}
 
-	AVDictionary *opts = NULL;
 	if (m->buffering && !m->is_local_file)
 		av_dict_set_int(&opts, "buffer_size", m->buffering, 0);
 
+	if ((ret = av_dict_parse_string(&opts, m->input_options,
+		"=", " ", 0))) {
+		blog(LOG_WARNING, "Failed to parse input demuxer settings: %s\n%s",
+			av_err2str(ret), m->input_options);
```
Comment (RytoEX) [inline deps/media-playback/media-playback/media.c]: Style:  Indent line continuations with two tabs.

File: plugins/obs-ffmpeg/obs-ffmpeg-source.c [hunk 2]
```diff
@@ -70,6 +71,8 @@ static bool is_local_file_modified(obs_properties_t *props,
 	obs_property_t *input = obs_properties_get(props, "input");
 	obs_property_t *input_format =obs_properties_get(props,
 			"input_format");
+	obs_property_t *input_options = obs_properties_get(props,
+		"input_options");
```
Comment (RytoEX) [inline plugins/obs-ffmpeg/obs-ffmpeg-source.c]: Style:  Indent line continuations with two tabs.

File: plugins/obs-ffmpeg/obs-ffmpeg-source.c [hunk 3]
```diff
@@ -160,6 +163,9 @@ static obs_properties_t *ffmpeg_source_getproperties(void *data)
 	obs_properties_add_text(props, "input_format",
 			obs_module_text("InputFormat"), OBS_TEXT_DEFAULT);
 
+	obs_properties_add_text(props, "input_options",
+		obs_module_text("InputOptions"), OBS_TEXT_DEFAULT);
```
Comment (RytoEX) [inline plugins/obs-ffmpeg/obs-ffmpeg-source.c]: Style:  Indent line continuations with two tabs.

File: plugins/obs-ffmpeg/obs-ffmpeg-source.c [hunk 4]
```diff
@@ -310,9 +320,12 @@ static void ffmpeg_source_update(void *data, obs_data_t *settings)
 
 		obs_source_set_async_unbuffered(s->source, false);
 	}
+	input_options = (char *)obs_data_get_string(settings,
+		"input_options");
```
Comment (RytoEX) [inline plugins/obs-ffmpeg/obs-ffmpeg-source.c]: Style:  Line breaks occur after 80 columns.   Lines 323-324 can be a single line.

File: deps/media-playback/media-playback/media.c (hunk 1 above)
Comment (pkviet) [reply deps/media-playback/media-playback/media.c]: done

File: plugins/obs-ffmpeg/obs-ffmpeg-source.c (hunk 2 above)
Comment (pkviet) [reply plugins/obs-ffmpeg/obs-ffmpeg-source.c]: done

File: plugins/obs-ffmpeg/obs-ffmpeg-source.c (hunk 3 above)
Comment (pkviet) [reply plugins/obs-ffmpeg/obs-ffmpeg-source.c]: done

File: plugins/obs-ffmpeg/obs-ffmpeg-source.c (hunk 4 above)
Comment (pkviet) [reply plugins/obs-ffmpeg/obs-ffmpeg-source.c]: done, thanks a lot !

**Comment by RytoEX** [comment]
I forgot to mention...

Please make sure to capitalize the first word of the commit subject, like this:

> obs-ffmpeg: Enable demuxer options for sources

**Comment by pkviet** [comment]
yes, Jim told me ! I corrected the title already as you can see above

**Comment by RytoEX** [comment]
You updated the PR title, but you still need to update the _commit subject_.  Those are separate.

**Comment by pkviet** [comment]
Oh, sorry. That's corrected now

**Pull request closed without merging**